internal/pkg/term/syncbuffer/termprinter.go

```diff
@@ -0,0 +1,169 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package syncbuffer provides a goroutine safe bytes.Buffer as well printing functionality to the terminal.
+package syncbuffer
+
+import (
+	"fmt"
+	"io"
+	"math"
+	"strings"
+
+	"github.com/aws/copilot-cli/internal/pkg/term/cursor"
+	"golang.org/x/term"
+)
+
+// printAllLinesInBuf represents to print the entire contents in the buffer.
+const (
+	printAllLinesInBuf = -1
+)
+
+// FileWriter is the interface to write to a file.
+type FileWriter interface {
+	io.Writer
+	Fd() uintptr
+}
+
+// LabeledTermPrinter is a printer to display label and logs to the terminal.
+type LabeledTermPrinter struct {
+	term             FileWriter           // term writes logs to the terminal FileWriter.
+	buffers          []*LabeledSyncBuffer // buffers stores logs before writing to the terminal.
+	numLines         int                  // number of lines that has to be written from each buffer.
+	padding          int                  // Leading spaces before rendering to terminal.
+	prevWrittenLines int                  // number of lines written from all the buffers.
+	termWidth        int                  // width of terminal.
+}
+
+// LabeledTermPrinterOption is a type alias to configure LabeledTermPrinter.
+type LabeledTermPrinterOption func(ltp *LabeledTermPrinter)
+
+// NewLabeledTermPrinter returns a LabeledTermPrinter that can print to the terminal filewriter from buffers.
+func NewLabeledTermPrinter(fw FileWriter, bufs []*LabeledSyncBuffer, opts ...LabeledTermPrinterOption) (*LabeledTermPrinter, error) {
+	width, err := terminalWidth(fw)
+	if err != nil {
+		return nil, fmt.Errorf("get terminal width: %w", err)
+	}
```

Comment:
shouldn't we calculate this every time `Print()` is called? in case someone changes the size of their terminal after the `LabeledTermPrinter` is created?

Reply:
That is a valid case. I will call the terminal width whenever we are printing the logs to terminal instead of initializing in the constructor.

Reply:
Done!